packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -176,7 +176,12 @@ class WebAssetServer implements AssetReader {
       if (testMode) {
         return server;
       }
-
+      
+      // In debug and profile builds allow rendering in a frame.
+      if (!buildInfo.mode.isRelease) {
```

Comment:
any reason not to remove on release builds too? It would probably be easier to test this if you removed the header on each request, see the existing devfs_web_test.dart

Reply:
Probably not many people will use flutter run to serve their production app, but it is not recommended to remove it as this header is used to avoid click-jacking attacks. 

I don't think is possible to remove a default header in a request. When creating a response you can add a map of headers but don't see how to remove one. Also, I can't override it as the opposite(eg X-Frame-Options: ALWAYS) doesn't exist.

`shelf.Response.ok(_kDefaultIndex, headers: headers);`

Reply:
> Probably not many people will use flutter run to serve their production app, but it is not recommended to remove it as this header is used to avoid click-jacking attacks.

But flutter run is still a debug mode essentially, its not a full app server. If it isn't safe for release mode it isn't safe for debug or profile mode.

Yeah, I'll admit I'm not familiar with the shelf APIs. This approach sounds fine as long as you test that the header isn't present somehow